lib/active_merchant/billing/gateways/redsys.rb

```diff
@@ -321,6 +323,25 @@ def add_payment(data, card)
         end
       end
 
+      def add_external_mpi_fields(data, options)
+        return unless options[:three_d_secure]
+
+        if options[:three_d_secure][:version]
+          data[:threeDSServerTransID] = options[:three_d_secure][:xid] if options[:three_d_secure][:xid]
+          data[:dsTransID] = options[:three_d_secure][:ds_transaction_id] if options[:three_d_secure][:ds_transaction_id]
+          data[:authenticacionValue] = options[:three_d_secure][:cavv] if options[:three_d_secure][:cavv]
+          data[:protocolVersion] = options[:three_d_secure][:version] if options[:three_d_secure][:version]
+
+          data[:authenticacionMethod] = options[:authentication_method] if options[:authentication_method]
+          data[:authenticacionType] = options[:authentication_type] if options[:authentication_type]
+          data[:authenticationFlow] = options[:authentication_flow] if options[:authentication_flow]
```

Comment:
is this actually `authenticationFlow` and inconsistent?

Reply:
I deliberately held-off on using the misspelling until the only place it's actually "needed", ie: actually talking to the endpoint. Should we consistently misspell it thru all the layers instead?

Reply:
Ah, though `build_merchant_mpi_external` only picks up `authenticacionFlow`, right? So I don't think it's actually getting passed. In any case I'd suggest consistency with the sibling fields here.